spring-integration-redis/src/main/java/org/springframework/integration/redis/inbound/RedisQueueMessageDrivenEndpoint.java

```diff
@@ -158,6 +160,16 @@ public void setRecoveryInterval(long recoveryInterval) {
 		this.recoveryInterval = recoveryInterval;
 	}
 
+	/**
+	 * Should data from the Redis queue be read from the left end of the queue (using leftPop)?
+	 * This is useful when the queue is fed by a software that uses rightPush.
```

Comment:
Yes, the case that I have in mind is a producing system that appends new messages to the right. I don't know what the traditional order is. If you use SI adapters on both sides, the messages in the queue are ordered oldest right, newest left. My producing system (logstash) orders reversely.

I now realized that this option is really agnostic to whether you change the reading side in order to use the list as a reversed FIFO queue, or as a  stack. So maybe my usage hints in the docs are too narrow-minded and should be removed.